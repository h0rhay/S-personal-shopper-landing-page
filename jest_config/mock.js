export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
