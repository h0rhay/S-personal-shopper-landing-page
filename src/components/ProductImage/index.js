import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const ProductImageComponent = styled.img`
  max-height: 18.75rem;
  object-fit: cover;
  display: block;
  position: relative;
  max-width: 21vmin;
`;

const ProductImage = ({ src, alt, className }) => (
  <ProductImageComponent loading="lazy" src={src} alt={alt} className={className} />
);

ProductImage.propTypes = {
  alt: PropTypes.string,
  src: PropTypes.string,
  className: PropTypes.string,
};

export default ProductImage;
