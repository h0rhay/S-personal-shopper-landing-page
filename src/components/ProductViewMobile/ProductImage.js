import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const ProductImageComponent = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: ${({ pos }) => (pos === 'mid' ? 'space-evenly' : pos === 'end' ? 'flex-end' : 'flex-start')};
  margin: 1rem 0;
`;

const ImageWrapper = styled.div`
  position: relative;
  width: 70%;
  &:before {
    content: '';
    position: absolute;
    top: ${({ pos }) => (pos === 'mid' ? '-10px' : pos === 'end' ? '-10px' : '-10px')};
    left: ${({ pos }) => (pos === 'mid' ? '-10px' : pos === 'end' ? '-10px' : '10px')};
    right: ${({ pos }) => (pos === 'mid' ? '10px' : pos === 'end' ? '10px' : '-10px')};
    bottom: ${({ pos }) => (pos === 'mid' ? '10px' : pos === 'end' ? '10px' : '10px')};
    border: ${({ pos }) => (pos === 'mid' ? '0px' : '1px')} solid ${({ theme }) => theme.accent};
    z-index: -1;
  }
`;

const ProductImage = ({ src, alt, pos = 'start' }) => (
  <ProductImageComponent pos={pos}>
    <ImageWrapper pos={pos}>
      <img loading="lazy" src={src} alt={alt} />
    </ImageWrapper>
  </ProductImageComponent>
);

ProductImage.propTypes = {
  alt: PropTypes.string,
  src: PropTypes.string,
  pos: PropTypes.oneOf(['start', 'mid', 'end']),
};

export default ProductImage;
