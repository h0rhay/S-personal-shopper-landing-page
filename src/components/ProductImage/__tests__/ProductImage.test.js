import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import ProductImage from '../index';

const props = {
  src: 'https://images.selfridges.com/is/image/selfridges/R03806044_CREAM_ALT01?$PDP_M_ZOOM$',
  alt: 'Concert wool midi dress',
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <ProductImage {...props} />
    </ThemeProvider>
  );

describe('ProductImage component', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme, props });
    expect(container).toMatchSnapshot();
  });
});
