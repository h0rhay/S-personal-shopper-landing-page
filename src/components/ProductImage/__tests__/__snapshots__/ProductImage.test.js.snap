// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage component should match screenshot 1`] = `
<div>
  <img
    alt="Concert wool midi dress"
    class="sc-bdvvtL bFuHx"
    loading="lazy"
    src="https://images.selfridges.com/is/image/selfridges/R03806044_CREAM_ALT01?$PDP_M_ZOOM$"
  />
</div>
`;
